Library/Homebrew/cask/cask.rb

```diff
@@ -281,6 +281,19 @@ def to_hash_with_variations
 
     private
 
+    def artifacts_list
+      artifacts.map do |artifact|
+        if artifact.is_a? Artifact::AbstractFlightBlock
+          { type: artifact.summarize }
+        else
+          {
+            type: artifact.class.dsl_key,
+            args: to_h_gsubs(artifact.to_args),
+          }
+        end
```

Comment:
```suggestion
        key, value = if artifact.is_a? Artifact::AbstractFlightBlock
          artifact.summarize
        else
          [artifact.class.dsl_key, to_h_gsubs(artifact.to_args)]
        end
        
        { key => value }
```
or
```suggestion
        key, value = if artifact.is_a? Artifact::AbstractFlightBlock
          [artifact.summarize, "AbstractFlightBlock"]
        else
          [artifact.class.dsl_key, to_h_gsubs(artifact.to_args)]
        end
        
        { key => value }
```

How about something like these?

Reply:
Yeah, good idea!